app/services/pairing_strategies/plain_pairing.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+# PlainPairing is a Plain Ruby model object that provides a Pairing representation
+# without the full overhead of the ActiveRecord Pairing object. This exists to keep
+# the pairing logic light on memory and object allocation / GC overhead.
+module PairingStrategies
+  class PlainPairing
+    def initialize(player1, player1_score, player2, player2_score)
+      @player1 = player1
+      @player1_score = player1_score
+      @player2 = player2
+      @player2_score = player2_score
+    end
+
+    attr_accessor :player1, :player1_score, :player2, :player2_score, :player1_side, :table_number
+
+    def players
+      [player1, player2]
+    end
```

Comment:
It looks like we're calling compact everywhere this is called. Maybe we should exclude nil from the result?

Reply:
sure.